reactor-support/src/main/java/tech/picnic/reactor/ReactorSpring4Util.java

```diff
@@ -0,0 +1,194 @@
+   * @return a function converting a Publisher to an SseEmitter
+   */
+  public static <T> Function<Publisher<? extends T>, SseEmitter> publisherToSse(
+      @Nullable MediaType mediaType, Duration heartbeat, LongFunction<T> heartbeatMapping) {
+    return publisherToSse(mediaType, heartbeat, heartbeatMapping, Schedulers.elastic());
+  }
+
+  @VisibleForTesting
+  static <T> Function<Publisher<? extends T>, SseEmitter> publisherToSse(
+      @Nullable MediaType mediaType,
+      Duration heartbeat,
+      LongFunction<T> heartbeatMapping,
+      Scheduler scheduler) {
+    return publisher ->
+        Flux.<T>from(publisher)
+            .publish(
+                flux ->
+                    flux.mergeWith(
+                        Flux.interval(Duration.ofMillis(heartbeat.toMillis()), scheduler)
+                            .map(heartbeatMapping::apply)
+                            .takeUntilOther(flux.ignoreElements())))
+            .as(publisherToSse(mediaType, NO_TIMEOUT));
+  }
+
+  /**
+   * Provides a function that subscribes to a {@link Publisher} and exposes its emissions as {@link
+   * SseEmitter server-sent events}.
+   *
+   * @param mediaType the media type of individual events; null indicates the context's default
+   *     media type
+   * @return a function converting a Publisher to an SseEmitter
+   */
+  public static Function<Publisher<?>, SseEmitter> publisherToSse(@Nullable MediaType mediaType) {
+    return publisherToSse(mediaType, new SseEmitter());
+  }
+
+  /**
+   * Provides a function that subscribes to a {@link Publisher} and exposes its emissions as {@link
+   * SseEmitter server-sent events}.
+   *
+   * @param timeout the SseEmitter's timeout
+   * @return a function converting a Publisher to an SseEmitter
+   */
+  public static Function<Publisher<?>, SseEmitter> publisherToSse(Duration timeout) {
+    return publisherToSse(null, timeout);
+  }
+
+  /**
+   * Provides a function that subscribes to a {@link Publisher} and exposes its emissions as {@link
+   * SseEmitter server-sent events}.
+   *
+   * @param mediaType the media type of individual events; null indicates the context's default
+   *     media type
+   * @param timeout the SseEmitter's timeout
+   * @return a function converting a Publisher to an SseEmitter
+   */
+  public static Function<Publisher<?>, SseEmitter> publisherToSse(
+      @Nullable MediaType mediaType, Duration timeout) {
+    return publisherToSse(mediaType, new SseEmitter(timeout.toMillis()));
+  }
+
+  private static Function<Publisher<?>, SseEmitter> publisherToSse(
+      @Nullable MediaType mediaType, SseEmitter sseEmitter) {
+    return publisher -> {
+      sseEmitter.onTimeout(
+          Flux.from(publisher)
+                  .subscribe(
+                      o -> {
+                        try {
+                          sseEmitter.send(o, mediaType);
+                        } catch (IOException e) {
+                          throw new UncheckedIOException(e);
```

Comment:
This exception must be explicitly caught here, where we don't need to do this in the `RxSpring4Util` equivalent, because this accepts a java util `Function` rather than the rx `Function` class